### PR TITLE
Big refactor: all things working

### PR DESCRIPTION
**Pull request by emi420**
This PR has all things working and serves as a starting point for the development of the library.

## Styling

This was one of the main issues, some styles were working on certain contexts (ex: web components Storybook) and failing in others (ex: React example). The main issue was with the styles provided by UnoCSS, utility classes like "flex" or "text-primary" were not always working. I've tried with Tailwind CSS but the issue persisted. The solution was to write the styles in CSS, this is something trivial as the utility classes are really basic.

Now you'll find that there's a style file for each component, for example`header.styles.ts`. 

We still use Class Variance Authority and make heavy use of CSS global variables, the ones provided by Shoelace prefixed with `--sl-` and some specific things prefixed with `--hot-`.

Variables are defined (or redefined for Shoelace) in two files: `theme/hot.css` and `theme/hot-sl.css`. I decided to use a single CSS file for color variables with the re-definition of Shoelace values because this library depends on it and it was not convenient to define variables twice. 

This solves #55 #53 #51 #50 #33 #34

## Logo component

This new component displays the HOT english logo (image and text) as default, using SVG files.

This solves #45 

## Storybook for React

It's really convenient for testing components in React while doing development, ensuring compatibility. In the future we could integrate both Storybooks, having an option for HTML and React, like the Shoelace docs.

For starting the React Storybook: `pnpm run dev-react`

This solves #54 

## Updated app examples

All examples were updated, not only in the app code but also the library dependency, using `git+https://github.com/hotosm/ui.git#dev` instead of a link. This is to make sure that the library is working in a separated app. You can even copy any example to a path outside the library, run `pnpm install && pnpm run dev` and it will work.

## Shoelace components

Shoelace web components are now exported without the `Sl` prefix, so you can import them like this:

```js
import { Toolbar, Button, Header, Logo } from '@hotosm/ui/components';
```

Same with the React ones:

```js
import { Toolbar, Button, Header, Logo } from '@hotosm/ui/components/react';
```

This makes things more consistent.

## Updated documentation

The `Readme.md` file was updated to be more clear and concise, some of its content will be moved to a detailed documentation later.

### CODE REVIEW

**Comment by emi420** [comment]
What we need to be working is styling and functionality, for both HOT custom components and Shoelace wrappers, not only when linking the package but also when installing it from command line using the repository URL (until we publish the package to the NPM registry), like this:

`pnpm install "git+https://github.com/hotosm/ui.git#dev`

I had to remove some dependencies and use a different strategy, specially for styling, because using neither UnoCSS nor Tailwind was not working when trying to install the package. The current approach is working, I've tested this branch in all the different contexts (Storybook, React Storybook, React, React 19, Svelte, Vue and HTML) and it's working on all of them.

Docs should be updated and some styles needs to be fixed before merging and continue working on components.

File: components/header/header.styles.ts [hunk 1]
```diff
@@ -0,0 +1,103 @@
+import { css } from 'lit';
+import { cva } from "class-variance-authority";
+
+export const headerVariants = cva(
```
Comment (JoltCode) [inline components/header/header.styles.ts:4]: I do feel at some point we should re-evaluate how we're doing styling, these files do seem a little hacky, perhaps we could look at https://panda-css.com/ or something similar in the future.

File: components/header/header.ts
Comment (JoltCode) [inline components/header/header.ts:1]: Small note for the future: we need to add prettier/biome/some sort of formatter, we need to make sure it's consistent between files and between editors.

File: components/index.ts [hunk 2]
```diff
@@ -200,67 +199,67 @@ export {
...
-    SlTab,
-    SlTabGroup,
-    SlTabPanel,
-    SlTag,
-    SlTextarea,
-    SlTooltip,
-    SlTree,
-    SlTreeItem,
-    SlVisuallyHidden,
+    SlAlert as Alert,
```
Comment (JoltCode) [inline components/index.ts:202]: Q: why are we having to type all these using "As"? Aren't the types of all these components not automatically inferred?

File: components/react/Header.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
+// @ts-ignore
```
Comment (JoltCode) [inline components/react/Header.ts:1]: Why are we ts-ignore'ing all the react files?

File: components/header/header.ts
Comment (spwoodcock) [reply components/header/header.ts:1]: We do have a pre-commit hook for prettier, but it's not enforcable.

I can enable the pre-commit.ci integration for this repo though to solve that 👍 (so it auto applies to all PRs)

File: README.md [hunk 4]
```diff
@@ -1,207 +1,145 @@
-# HOT Shared UI
-
-<!-- markdownlint-disable -->
-<p align="center">
```
Comment (spwoodcock) [inline README.md:4]: I preferred this centralised layout, as it aligns with READMEs in our other tools too

**Review by spwoodcock — APPROVED** [review]
Love it! Great work 🙌

As Joe said, we can possibly refine some extra things together in the future, but the main thing is right now that this works nicely and we can start designing new components 😃

**Comment by spwoodcock** [comment]
Should we remove the React components as they are unnecessary overhead?

Plus the index `import xxx as xxx` can.be looked at as Joe commented. 

Otherwise, we are set to merge 👍

File: components/header/header.styles.ts (hunk 1 above)
Comment (emi420) [reply components/header/header.styles.ts:4]: The current approach is pretty similar of what Shoelace uses, for example:

https://github.com/shoelace-style/shoelace/blob/next/src/components/button/button.styles.ts

Don't you think Panda CSS would introduce more confusion and more code to maintain? currently we're using common CSS, with global variables defined by Shoelace and some extras by us.

File: components/header/header.ts
Comment (emi420) [reply components/header/header.ts:1]: 100% agree with this, we should setup a linting config for using in VSCode at least.

File: components/index.ts (hunk 2 above)
Comment (emi420) [reply components/index.ts:202]: I used "as" in order to have all components with common names, without the "Sl" prefix.

I'm not sure if the final user will have the need to distinguish between pure Shoelace-based components and other ones.

If we use names without prefixes, we can replace any Shoelace component, like Button, with our own version if that's needed, or wrap them and add functionality, or add new components ...

File: components/react/Header.ts (hunk 3 above)
Comment (emi420) [reply components/react/Header.ts:1]: Oh, that line was there before and I forgot to check why it was there.

I don't think we should delete the React components, even Shoelace have React support. Now that we have everything working you just need to add a wrapper with `createComponent` and an export in `components/react/index.js`.

File: README.md (hunk 4 above)
Comment (emi420) [reply README.md:4]: Yes? it's not very clean ..

File: README.md [hunk 5]
```diff
@@ -1,207 +1,145 @@
...
-      <img src="https://img.shields.io/github/license/hotosm/ui.svg" alt="License">
-  </a>
-</p>
+# HOT UI
 
-📖 **Documentation**: <a href="https://hotosm.github.io/ui/" target="_blank">https://hotosm.github.io/ui/</a>
+## Shared UI components with HOT theming
 
-🖥️ **Source Code**: <a href="https://github.com/hotosm/ui" target="_blank">https://github.com/hotosm/ui</a>
+HOT themed UI components to reduce code duplication and make live easier for developers, available as [Web Components](https://developer.mozilla.org/en-US/docs/Web/API/Web_components) and witn first-class React support
```
[the remark this answers is not in the document]
Comment (emi420) [reply README.md:5]: But HOT UI is also working in React 18

File: README.md (hunk 4 above)
Comment (spwoodcock) [reply README.md:4]: Clean based on what metric?

Sure it's HTML in Markdown which isn't ideal, but it makes for a nicer looking README file so it's worth it.

So many other projects use this approach 😅

File: components/header/header.styles.ts (hunk 1 above)
Comment (spwoodcock) [reply components/header/header.styles.ts:4]: I do quite like the simplicity of just CSS actually 👍

But let's sit and reevaluate at some point in the future to see if there is any refinement that could be done

File: components/index.ts (hunk 2 above)
Comment (spwoodcock) [reply components/index.ts:202]: Fair point - it makes it easier for the end user at the expense of a little confusion for dev contributors - two competing concerns

File: components/header/header.styles.ts (hunk 1 above)
Comment (JoltCode) [reply components/header/header.styles.ts:4]: For sure, just something to think about and evaluate in the future! 👍

File: README.md (hunk 4 above)
Comment (emi420) [reply README.md:4]: I mean, I don't think the design and content organization is clean, using HTML is ok. IMO it should be easier to understand, with more consistent alignment and without excessive amount of information.

**Pull request merged**